### DISCUSSION TITLE
[Proposal] Relax the need for brackets for one-line try-catch-finally statements.

### DISCUSSION THREAD

**Discussion by scott-parkhill** [dotnet/csharplang#5982] [Language Ideas] [answered]
Currently in the language, try-catch blocks are very unwieldy when used with single statements. They take up many lines of code following the Allman brace style, and all those lines of code effectively pollute what is happening and reduce readability. For a try-catch statement where the try and catch are each executing one statement, it takes 8 lines of code to effectively execute two statements.

My proposal is to give the try-catch, and by extension the try-catch-finally, C if-style bracing. That is, we currently have to do the following (using a dummy example of the first thing that came to mind):

```csharp
try
{
    await _dbContext.SaveChangesAsync();
}
catch (DbUpdateException)
{
    throw;
}
finally
{
    _dbContext?.Dispose();
}
```

My proposal is to allow the following to be valid C# code:

```csharp
try
    await _dbContext.SaveChangesAsync();
catch (DbUpdateException)
    throw;
finally
    _dbContext?.Dispose();
```

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/4676

**Reply by scott-parkhill** [reply]
Very interesting discussion, thank you for pointing me towards this.

**Comment by HaloFour** [comment]
TL;DR, the reason C# doesn't allow it is the same reason C++ doesn't allow it, it creates ambiguity given `try` can be nested and have any number of `catch` clauses.
